types/src/sample.rs

```diff
@@ -111,20 +110,14 @@ impl Sample {
             AxisType::Col => (index % square_len, index / square_len),
         };
 
-        let mut tree = Nmt::with_hasher(NamespacedSha2Hasher::with_ignore_max_ns(true));
-
-        let shares = eds.axis(axis_type, axis_index)?;
-        let (data_shares, parity_shares) = shares.split_at(square_len / 2);
+        let (row_index, column_index) = match axis_type {
+            AxisType::Row => (axis_index, sample_index),
+            AxisType::Col => (sample_index, axis_index),
+        };
```

Comment:
So this reverses the indexing from the previous match, making both of them pointless. If we want to have this, then we could remove both matches and just have
```
let (row_index, column_index) = (index / square_len, index % square_len);
```

Tho I think it was correct before and now it's not, because now axis doesn't matter at all

Reply:
I will clean this up on my next PR

Reply:
but shouldn't this be removed from this PR?

Reply:
This reverses the indexing from the previous match, but on separate variables. `eds.axis_nmt` uses `axis_index`, `build_range_proof` uses `sample_index` to index along the axis; while `row/col_index` are used just for getting the share. 
So I think while mixing the two indexing methods is a bit misleading, it's still correct. Not sure how to eliminate that cleanly.

Reply:
Oh and original bug was `shares[sample_index]` rather than intended `shares[index]`, it should have used "global" index, rather than offset along the axis. 🙈

Reply:
> Not sure how to eliminate that cleanly.

I didn't open a PR yet, but this is how I did it: https://github.com/eigerco/lumina/commit/c43ca77b84c58eae7883ff4546939206e28c8281

Reply:
Also I think the code of this PR is correct because `axis_nmt` was reversing indexes again in case of `AxisType::Col`.